src/Pdo/PdoSelectCapableTrait.php

```diff
@@ -0,0 +1,135 @@
+<?php
+
+namespace RebelCode\Storage\Resource\Pdo;
+
+use Dhii\Config\Exception\ConfigExceptionInterface;
+use Dhii\Expression\ExpressionInterface;
+use Dhii\Expression\LogicalExpressionInterface;
+use Dhii\Util\String\StringableInterface as Stringable;
+use PDO;
+use PDOStatement;
+use Psr\Container\NotFoundExceptionInterface;
+use Traversable;
+
+/**
+ * Common functionality for objects that can retrieve records from a database using PDO.
+ *
+ * @since [*next-version*]
+ */
+trait PdoSelectCapableTrait
+{
+    /**
+     * Executes a SELECT SQL query, retrieving records from the database that satisfy the given condition.
+     *
+     * @since [*next-version*]
+     *
+     * @param LogicalExpressionInterface|null $condition Optional condition that records must satisfy.
+     *                                                   If null, all records in the target table are retrieved.
+     *
+     * @throws ConfigExceptionInterface   If an error occurred while reading from the config.
+     * @throws NotFoundExceptionInterface If required configuration was not found in the config.
+     *
+     * @return array|Traversable A list of retrieved records.
+     */
+    protected function _select(LogicalExpressionInterface $condition = null)
+    {
+        $fields       = $this->_getSqlSelectFieldNames();
+        $valueHashMap = ($condition !== null)
+            ? $this->_getPdoExpressionHashMap($condition, $fields)
+            : [];
+
+        $query = $this->_buildSelectSql(
+            $this->_getSqlSelectColumns(),
+            $this->_getSqlSelectTables(),
+            $this->_getSqlSelectJoinConditions(),
+            $condition,
+            $valueHashMap
+        );
+
+        return $this->_executePdoQuery($query, $valueHashMap)->fetchAll(PDO::FETCH_ASSOC);
+    }
+
+    /**
+     * Builds a SELECT SQL query.
+     *
+     * @since [*next-version*]
+     *
+     * @param string[]|Stringable[]           $columns        A list of names of columns to select.
```

Comment:
In such cases, should we perhaps also allow `Traversable`?

Reply:
Yes, good idea. `Traversable` fits with the way this parameter is being used.